tasks/main.yml

```diff
@@ -1,32 +1,29 @@
 ---
 # tasks file for cyhy_core
 
-#
-# Grab the cyhy-core code
-#
-- name: Create the /var/local/cyhy/core directory
-  file:
-    mode: 0755
-    path: /var/local/cyhy/core
-    state: directory
-
-- name: Download and untar the cyhy-core tarball
-  # ansible-lint E208 gives an error if we don't specify the mode,
-  # but we definitely don't want to do that here since it will
-  # override the permissions on the files in the archive.
-  unarchive: # noqa 208
-    src: "https://api.github.com/repos/cisagov/cyhy-core/tarball/develop"
-    dest: /var/local/cyhy/core
-    remote_src: yes
-    extra_opts:
-      - "--strip-components=1"
+- name: Install system versions of the Python packages that cyhy-core needs
+  package:
+    name:
+      # PyCrypto
+      - python-crypto
+      - python-dateutil
+      - python-docopt
+      - python-geoip2
+      - python-maxminddb
+      - python-netaddr
+      - python-pandas
+      - python-progressbar
+      # The version available for Debian Stretch is 3.4.0, and the cyhy-core
+      # package currently has a requirement of < 3. When the cyhy-core package
+      # supports a more recent version of PyMongo, this should be enabled.
+      # - python-pymongo
+      - python-six
+      # PyYAML
+      - python-yaml
 
-#
-# Install cyhy-core
-#
-- name: Install cyhy-core
+- name: Install the cyhy-core package
   pip:
-    name: file:///var/local/cyhy/core
+    name: https://api.github.com/repos/cisagov/cyhy-core/tarball/develop
```

Comment:
For maximum flexibility, I recommend adding an optional `virtualenv` parameter to this role and setting [the `virtualenv` parameter of the `pip` Ansible module](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/pip_module.html#parameter-virtualenv) equal to `"{{ virtualenv | default(omit) }}"`, similar to what I do [in cisagov/ansible-role-assessment-tool](https://github.com/cisagov/ansible-role-assessment-tool/blob/develop/tasks/install_tool.yml#L13).

I think in the future we will take advantage of this parameter, since *we must get away from installing packages via `pip` under the system python and using a melange of system python packages and `pip` packages*.  Note that, since you are using the system python here, this change will also require that you install the `python-virtualenv` package.

I don't think it will make a difference here, since Ansible is selecting python2 wherever this role is applied, but you may want to also add an optional `virtualenv_python` host parameter to match [the one in the `pip` Ansible module](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/pip_module.html#parameter-virtualenv_python).

Reply:
To be clear, I am fine with _not_ making this change in this PR, as long as we create an issue or issues for:
* Changing CyHy in general to use a venv _or_ a Docker container (probably a cisagov/cyhy-system issue), **and**
* Making cisagov/cyhy-core install everything it needs via `pip` packages listed in its `setup.py` or `requirements.txt` and no longer relying on pre-installed system python packages (maybe another cisagov/cool-system issue, since there are several CyHy repos that need this kind of reworking)

I'm getting religious about the system Python being for the system and everything else being in a Docker container or using a venv.  I _think_ @mcdonnnj is with me on this crusade.  At the same time, I realize that we have several balls in the air right now and Rome wasn't built in a day.

Reply:
I _am_ on board with this crusade, but I do not think this role and package should implement these changes. I think the scripts found in [`bin/`](https://github.com/cisagov/cyhy-core/tree/develop/bin) of [cisagov/cyhy-core](https://github.com/cisagov/cyhy-core) should be broken out into their own repository. Then every package that relies on the `cyhy-core` package would install it into the package's virtual environment after appropriate updates. This would eliminate the need for this role entirely, so it would be archived.

I think an issue in [cisagov/cyhy-system](https://github.com/cisagov/cyhy-system) encapsulating that would be appropriate, but what are your thoughts?

Reply:
I am fine with that.

Reply:
I am also in agreement with this plan.

Reply:
@dav3r @jsf9k Does https://github.com/cisagov/cyhy-system/issues/14 encapsulate this discussion sufficiently?

Reply:
That's perfect.  Thank you for creating it.

Reply:
I have also added https://github.com/cisagov/cyhy-core/issues/57 to track the decoupling I mentioned in the appropriate project.